### DISCUSSION TITLE
endpoint_url not working as intended

### DISCUSSION THREAD

**Discussion by dfencegit** [quay/quay#1423] [Ideas]
Hi,

The addition of the ability to configure an endpoint_url value for Azure storage backend is a welcome update as it allows targetting of non-public Azure storage backends however it does not work as intended due to the way the base_client.py module from the Azure SDK for python works.

In order for the endpoint_url value to work correctly for non-public Azure endpoints the credential passed in to the BlobServerClient call needs to include the account name.

A rough patch to the "azurestorage.py" would be as below:

        CREDENTIAL = { "account_name": self._azure_account_name, "account_key": self._azure_account_key }

        self._blob_service_client = BlobServiceClient(
            self._azure_connection_string
            if self._azure_connection_string
            else AZURE_STORAGE_URL_STRING["global"].format(self._azure_account_name),
            credential=CREDENTIAL,
        )

**Comment by dmesser** [comment]
The initial use case for the endpoint_url was to enable Microsoft Azure Government regions which have a different, but public storage endpoint. Sounds like we could easily introduce non-public Azure backends as well, while we are at it.

**Reply by dfencegit** [reply]
Yes would seem to be a simple change to facilitate this and would be really useful for us.  How do we go about getting this put in place?

**Reply by dmesser** [reply]
We can see if some of the maintainers can pick it up, but we are also happy about any contributions :)

**Reply by dfencegit** [reply]
I'm definately not a developer so I suspect you would not want any code I would write :-)
